packages/gatsby-plugin-mdx/utils/gen-mdx.js

```diff
@@ -116,6 +116,10 @@ export const _frontmatter = ${JSON.stringify(data)}`
       reporter,
       cache,
       pathPrefix,
+      compiler: {
+        parseString: compiler.parse.bind(compiler),
+        generateHTML: ast => mdx.sync(ast, options),
```

Comment:
I don't have enough context on this but it looks like `generateHTML` here is `sync` while in `gatsby-transformer-remark` it is `async`:

https://github.com/gatsbyjs/gatsby/blob/ca19076f2964c0f6101d28b3a68a117b33d8cc70/packages/gatsby-transformer-remark/src/extend-node-type.js#L184-L187

(note the `getHTML` function is async)

It probably caused #20000 here:

https://github.com/gatsbyjs/gatsby/blob/ca19076f2964c0f6101d28b3a68a117b33d8cc70/packages/gatsby-remark-images/src/index.js#L118

(as return value is expected to be sync, but we have a Promise here in case of `gatsby-transformer-remark`)

Reply:
good catch